### DISCUSSION TITLE
Cannot inject @Repository annotated interface from external jar into @MicronautTest

### DISCUSSION THREAD

**Discussion by gbennardo** [micronaut-projects/micronaut-data#3011] [Q&A]
Hi everyone and thanks in advance for your support, I'm trying to build a Micronaut library in order to access database and to avoid repeated code in various projects.

I'm able to Inject `@Singleton` and `@Repository` annotatated classes from the external Micronaut library to a given project using `@Import` annotation when I run the application but when I try to inject into a `@MicronautTest` annotated class there is no bean candidate to inject 
```
01:36:22.848 [Test worker] DEBUG i.m.context.DefaultBeanContext - Finding candidate beans for instance: com.example.project.TestClass@5b1e88f
01:36:22.854 [Test worker] DEBUG i.m.context.DefaultBeanContext - Finding candidate beans for type: TestRepository
01:36:22.854 [Test worker] DEBUG i.m.context.DefaultBeanContext - Resolved bean candidates [] for type: TestRepository

org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter TestRepository
```

Annotation processors to compile repository jar

```
<annotationProcessorPaths>
    <path>
        <groupId>org.projectlombok</groupId>
        <artifactId>lombok</artifactId>
        <version>1.18.34</version>
    </path>
    <path>
        <groupId>io.micronaut</groupId>
        <artifactId>micronaut-inject-java</artifactId>
        <version>${micronaut.version}</version>
    </path>
    <path>
        <groupId>io.micronaut.data</groupId>
        <artifactId>micronaut-data-processor</artifactId>
        <version>${micronaut.data.version}</version>
    </path>
    <path>
        <groupId>io.micronaut.data</groupId>
        <artifactId>micronaut-data-model</artifactId>
        <version>${micronaut.data.version}</version>
    </path>
</annotationProcessorPaths>
```

Metadata included into repository jar:

```
$TestRepository$Intercepted$Definition.class
$TestRepository$Intercepted$Definition$Exec.class
$TestRepository$Intercepted$Definition$Reference.class
```

**Comment by dstepanov** [comment]
I believe `@Import` works only with simple beans. The best is to apply the Micronaut Data annotation proccessor to your libary classpath so metadata are included in the library.

**Reply by gbennardo** [reply]
The weird thing is that if I run the application (where I import the repository) it works. If I run any `@MicronautTest` in the same application the repository is no more injected.

All metadata are included into the jar

I updated my original question with annotation processors and metadata included into the repository jar

**Reply by dstepanov** [reply]
Maybe somethign is missconfigured, please create a sample app to check
